### DISCUSSION TITLE
`ElTree` suggestion

### DISCUSSION THREAD

**Discussion by xiaoxian521** [element-plus/element-plus#910] [Ideas]
功能：树组件可以扩展成支持大数据渲染以及搜索功能

场景：这种功能在比较大的企业中是很常见的，比如公交行业有很多车辆，自然车上的摄像机就少不了，数据量可达10-100w

问题：那么在使用这种Dom直接渲染大量数据的方式就会很不友好了，可采用虚拟列表这种方式只加载当前页面的Dom，
那么右键菜单的需求又来了，树节点可能不光光只是渲染数据，还应该支持更多的交互操作，比如在某个节点上右键，需要弹出一个菜单，菜单上是一些操作，比如打开摄像机，或者收藏节点等功能。

**Comment by jw-foss** [comment]
I am working on implementing `VirtualList` but to integrate it into components could have impact on the existing ones. Once it's finished I will export it as a complete component for user to use, but the process should take time.

Also something I wanted to mention here is that the whole crew is not full time developer on developing this project, so it might take even longer than usual because we are under the situation of lacking resources.

**Comment by xiaoxian521** [comment]
如果支持大数据的话  是整体组件都支持 还是 只是提供一个配置属性 去支持呢？

**Reply by jw-foss** [reply]
As I mentioned it should be a continuous process, with smooth transition I suppose it needs to be separated at the beginning and merged together.

**Reply by xiaoxian521** [reply]
haha It's really a formidable function support you
